08-translations.Rmd

```diff
@@ -0,0 +1,160 @@
+different ways to pluralize a quantity.
+
+In .po files, this shows up in the form of `msgid_plural` entries, followed by several ordered `msgstr` entries. Here's an example from
+[`R-de.po`](https://github.com/r-devel/r-svn/blob/c715d61cb74b3fee2d035faed9b258e86e420b75/src/library/base/po/R-de.po#L2015-L2018):
+
+```
+msgid "Warning message:\n"
+msgid_plural "Warning messages:\n"
+msgstr[0] "Warnmeldung:\n"
+msgstr[1] "Warnmeldungen:\n"
+```
+
+The two entries in English correspond to the singular and plural messages; the two entries in German correspond similarly, because
+pluralization rules in German are similar to those in English. The situation in Lithuanian
+([`R-lt.po`](https://github.com/r-devel/r-svn/blob/c715d61cb74b3fee2d035faed9b258e86e420b75/src/library/base/po/R-lt.po#L1999-L2003))
+is more divergent:
+
+```
+msgid "Warning message:\n"
+msgid_plural "Warning messages:\n"
+msgstr[0] "Įspėjantis pranešimas:\n"
+msgstr[1] "Įspėjantys pranešimai:\n"
+msgstr[2] "Įspėjančių pranešimų:\n"
+```
+
+This corresponds to the 3 different ways to pluralize words in Polish.
+
+What do `0`, `1`, and `2` correspond to, exactly? Ideally, this will be clear to native speakers of the language, but for
+clarity, it is the solution to a small arithmetic problem that can be found in the language's metadata entry. Look for the
+`Plural-Forms` entry in the metadata at the top of the .po file;
+[here it is for Lithuanian](https://github.com/r-devel/r-svn/blob/c715d61cb74b3fee2d035faed9b258e86e420b75/src/library/base/po/R-lt.po#L18-L19):
+
+```
+"Plural-Forms: nplurals=3; plural=(n%10==1 && n%100!=11 ? 0 : n%10>=2 && (n"
+"%100<10 || n%100>=20) ? 1 : 2);\n"
+```
+
+`nplurals` tells us how many entries correspond to each `msgid_plural` for this language. `plural` tells us, for the
+quantity `n`, which entry to use. The arithmetic is C code; most important if you really want to parse this and are only
+familiar with R code is C's [ternary operator](https://en.wikipedia.org/wiki/%3F:): `test ? valueIfTrue : valueIfFalse`
+is a handy way to write R's `if (test) valueIfTrue else valueIfFalse`.
+
+Parsing, we get the following associations:
+
+ - the `0` entry corresponds to when a number equals 1 modulo 10 (i.e., 1, 11, 21, 31, ...) _except_ numbers equaling
+   11 modulo 100 (i.e., 11, 111, 211, 311, ...). Combining, that's 1, 21, 31, ..., 91, 101, 121, 131, ..., 191, ...
+ - the `1` entry corresponds to numbers at least 2 modulo 10 (2, 3, ..., 8, 9, 12, 13, 14, ...) and _either_
+   below 10 modulo 100 (0, 1, ..., 9, 100, 101, ..., 109, ...) _or_ exceeding 20 modulo 100 (21, 22, ..., 99). Combining,
+   that's 2, 3, ..., 9, 22, 23, ..., 29, 32, 33, ... 39, ..., 102, 103, ..., 109, 122, 123, ...
+ - The `2` entry corresponds to all other numbers, i.e. 0, 10, 11, 12, ..., 19, 20, 30, ..., 90, 100, 110, 111, 112, ...
+
+TODO(michaelchirico): How to discover a new `nplurals/plural`.
```

Comment:
Can we put these TODOS in HTML comments?

Reply:
Yep, done. I left them as visible TODOs with the intention of finishing them before merging, but if we've passed MVP we can just hide them for now.